Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -167,3 +167,176 @@ extension UnsafeMutablePointer where Pointee == CTensorHandle? {
         self.init(unwrapped)
     }
 }
+
+//===------------------------------------------------------------------------------------------===//
+// Hashing
+//===------------------------------------------------------------------------------------------===//
+
+internal extension FixedWidthInteger {
+    @usableFromInline
+    init(fromBytes bytes: ArraySlice<UInt8>, startingAt index: Int) {
+        if bytes.isEmpty { self = 0; return }
+        let count = bytes.count
+        self = Self(0)
+        for i in 0..<MemoryLayout<Self>.size {
+            let j = (MemoryLayout<Self>.size - i - 1) * 8
+            self |= count > 0 ? Self(bytes[index.advanced(by: i)]) << j : 0
+        }
+    }
+
+    @usableFromInline
+    func bytes(count byteCount: Int = MemoryLayout<Self>.size) -> [UInt8] {
+        let valuePointer = UnsafeMutablePointer<Self>.allocate(capacity: 1)
```

Comment:
I'd suggest using a local variable here instead of a heap allocation.

```swift
var value = littleEndian
```

Then you can pass pointers to it as `&value`. Also, there's probably a more descriptive name than "value".